test/app-tap/datetime.test.lua

```diff
@@ -523,6 +528,323 @@ test:test("__index functions()", function(test)
+
+local function catchadd(A, B)
+    return pcall(function() return A + B end)
+end
+
+--[[
+Matrix of addition operands eligibility and their result type
+
+|                 |  datetime | interval |
++-----------------+-----------+----------+
+| datetime        |           | datetime |
+| interval        |  datetime | interval |
+]]
+test:test("Matrix of allowed time and interval additions", function(test)
+    test:plan(32)
+
+    -- check arithmetic with leap dates
+    local T1970 = date.new{year = 1970, month = 1, day = 1}
+    local T2000 = date.new{year = 2000, month = 1, day = 1}
+    local I1 = date.interval.new{day = 1}
+    local M2 = date.interval.new{month = 2}
+    local M10 = date.interval.new{month = 10}
+    local Y1 = date.interval.new{year = 1}
+    local Y5 = date.interval.new{year = 5}
+
+    test:is(catchadd(T1970, I1), true, "status: T + I")
+    test:is(catchadd(T1970, M2), true, "status: T + M")
+    test:is(catchadd(T1970, Y1), true, "status: T + Y")
+    test:is(catchadd(T1970, T2000), false, "status: T + T")
+    test:is(catchadd(I1, T1970), true, "status: I + T")
+    test:is(catchadd(M2, T1970), true, "status: M + T")
+    test:is(catchadd(Y1, T1970), true, "status: Y + T")
+    test:is(catchadd(I1, Y1), true, "status: I + Y")
+    test:is(catchadd(M2, Y1), true, "status: M + Y")
+    test:is(catchadd(I1, Y1), true, "status: I + Y")
+    test:is(catchadd(Y5, M10), true, "status: Y + M")
+    test:is(catchadd(Y5, I1), true, "status: Y + I")
+    test:is(catchadd(Y5, Y1), true, "status: Y + Y")
+
+    test:is(tostring(T1970 + I1), "1970-01-02T00:00:00Z", "value: T + I")
+    test:is(tostring(T1970 + M2), "1970-03-01T00:00:00Z", "value: T + M")
+    test:is(tostring(T1970 + Y1), "1971-01-01T00:00:00Z", "value: T + Y")
+    test:is(tostring(I1 + T1970), "1970-01-02T00:00:00Z", "value: I + T")
+    test:is(tostring(M2 + T1970), "1970-03-01T00:00:00Z", "value: M + T")
+    test:is(tostring(Y1 + T1970), "1971-01-01T00:00:00Z", "value: Y + T")
+    test:is(tostring(Y5 + Y1), "+6 years", "Y + Y")
+
+    assert_raises_like(test, expected_interval_but,
+                       function() return T1970 + 123 end)
+    assert_raises_like(test, expected_interval_but,
+                       function() return T1970 + {} end)
+    assert_raises_like(test, expected_interval_but,
+                       function() return T1970 + "0" end)
+
+    local max_date = date.new{year = MAX_DATE_YEAR - 1}
+    local new_ival = date.interval.new
+    test:is(catchadd(max_date, new_ival{year = 1}), true, "max + 1y")
+    test:is(catchadd(max_date, new_ival{year = 2}), false, "max + 2y")
+    test:is(catchadd(max_date, new_ival{year = 10}), false, "max + 10y")
+    test:is(catchadd(max_date, new_ival{month = 1}), true, "max + 1m")
+    test:is(catchadd(max_date, new_ival{month = 6}), true, "max + 6m")
+    test:is(catchadd(max_date, new_ival{month = 12}), true, "max + 12m")
+    test:is(catchadd(max_date, new_ival{month = 24}), false, "max + 12m")
```

Comment:
`12m` -> `24m`.

Reply:
Fixed (here with `add` and also elsewhere with `sub`)